src/base/bittorrent/session.h

```diff
@@ -50,10 +50,14 @@
 #include "sessionstatus.h"
 #include "torrentinfo.h"
 
-#if ((LIBTORRENT_VERSION_NUM >= 10206) && !defined(Q_OS_WIN))
+#if ((LIBTORRENT_VERSION_NUM >= 10206) && !defined(Q_OS_WIN)) || (LIBTORRENT_VERSION_NUM >= 10212)
 #define HAS_HTTPS_TRACKER_VALIDATION
 #endif
 
+#if ((10212 <= LIBTORRENT_VERSION_NUM) && (LIBTORRENT_VERSION_NUM < 20000)) || (LIBTORRENT_VERSION_NUM >= 20002)
```

Comment:
This tripped me up the first time I read it. I would suggest `#if ((LIBTORRENT_VERSION_NUM >= 10212) && (LIBTORRENT_VERSION_NUM < 20000)) || (LIBTORRENT_VERSION_NUM >= 20002)` instead.

Reply:
I don't think so, this form was chosen to mimic mathematical bound: `[0 < x < 10]`.

Reply:
In an ideal world this would be possible to do directly with preprocessor directives. But in C++ and other programming languages, my brain is hardwired to expect the form `x > 0 && x < 10` for expressing mathematical bounds as standard, and I suspect this is the case for many other people (at least from personal/anecdotal experience). But hey, if others don't oppose this, I don't consider this a blocker.

Reply:
I agree with @FranciscoPombal if it matters.

Reply:
If it really matters then it would have to be written in the coding guidelines and that is for another PR, I don't wish to stay on this topic any longer.

Reply:
@glassez thoughts?